### DISCUSSION TITLE
uses system eigen?

### DISCUSSION THREAD

**Discussion by LuckyTurtleDev** [libigl/libigl#2002] [Q&A] [answered]
Hi it is possible to you the eigen package, witch is already installed on the system?
I am using already the following cmake option:
```
cmake -DCMAKE_INSTALL_PREFIX=/usr \
		-DLIBIGL_BUILD_TESTS=OFF \
		-DLIBIGL_BUILD_TUTORIALS=OFF \
		-DLIBIGL_SKIP_DOWNLOAD=ON \
		-DLIBIGL_USE_STATIC_LIBRARY=OFF \
		-DLIBIGL_COPYLEFT_CGAL=OFF \
		-DLIBIGL_COPYLEFT_COMISO=OFF \
		-DLIBIGL_EMBREE=OFF \
		-DLIBIGL_RESTRICTED_MATLAB=OFF \
		-DLIBIGL_RESTRICTED_MOSEK=OFF \
		-DLIBIGL_OPENGL=OFF \
		-DLIBIGL_GLFW=OFF \
		-DLIBIGL_IMGUI=OFF \
		-DLIBIGL_PNG=OFF \
		-DLIBIGL_PREDICATES=OFF \
		-DLIBIGL_COPYLEFT_TETGEN=OFF \
		-DLIBIGL_RESTRICTED_TRIANGLE=OFF \
```
Sadly I have not found an option to disable eigen: https://github.com/libigl/libigl/blob/5da64bcbb581d189bef4f3b5cb503e03e0d94d99/CMakeLists.txt#L100-L119

Cmake does also install header to `/include/Eigen`, with are already provided by the eigen package.

**Comment by sl1pkn07** [comment, marked as answer]
my approach is set

~~~
		-DFETCHCONTENT_SOURCE_DIR_EIGEN=/usr/include/eigen3 \
		-DFETCHCONTENT_UPDATES_DISCONNECTED=ON \
~~~

and remove the eigen .cmake files and libigl eigen bundled headers after install. and let to cmake find the system eigen for use with libigl

because system eigen cmakefiles exports `Eigen3::Eigen`

~~~
┌─┤[$]|[sl1pkn07]|[sL1pKn07]|[/tmp/makepkg/libigl/pkg/libigl/usr/lib/cmake/igl]|
└───╼  grep -R Eigen
LibiglConfigTargets.cmake:  INTERFACE_LINK_LIBRARIES "Eigen3::Eigen;Threads::Threads"
LibiglConfigTargets.cmake:foreach(_target "Eigen3::Eigen" )
~~~

~~~
└───╼  cat /usr/share/eigen3/cmake/Eigen3Config.cmake 
# This file exports the Eigen3::Eigen CMake target which should be passed to the
# target_link_libraries command.
~~~

**Reply by VorpalBlade** [reply]
While this sounds like it would work, it seems to me, it would better to fix this upstream with a suitable cmake option. It is always best to strive to avoid downstream patching.

**Reply by LuckyTurtleDev** [reply]
I have apply this change, I hope their will no more issues anymore.

Of course fixing this upstream would be a good  idea, but this would not effect the current already released libigl version.

**Comment by sl1pkn07** [comment]
yep. better in upstream than patch in local

**Comment by alecjacobson** [comment]
Agreed. Happy to merge a suitable PR. (Obviously, the user introduces all risks of using a mismatched eigen version, so this would still be officially discouraged)

**Comment by jdumas** [comment]
In general I would recommend against installing libigl manually in your system. You should install libigl via a package manager. Libigl is already available through [hunter](https://hunter.readthedocs.io/en/latest/packages/pkg/libigl.html), [conan](https://conan.io/center/libigl) and [vcpkg](https://vcpkg.info/port/libigl).

- If you just want to compile libigl with the system Eigen, providing `FETCHCONTENT_SOURCE_DIR_EIGEN` as a compile option works well enough.
- If you are compiling a subproject that depends on libigl, you are free to provide your own `Eigen3::Eigen` target before including libigl.
- If you want to install libigl on your system, I would recommend using a package manager.

**Reply by VorpalBlade** [reply]
This question was raised in the context of the packaging for Arch Linux (AUR), so recommending to use the distro package doesn't really help here

 It is needed as a dependency of some other packages (such as superslicer): https://aur.archlinux.org/packages/libigl

**Reply by jdumas** [reply]
Sure. Building and installing libigl via AUR makes sense. I just wanted to make sure we're not encouraging "manual" installation into a user's sytem.
